Editor/ReactiveObjects/MeshFiltering/VertexFilterByBone.cs

```diff
@@ -0,0 +1,106 @@
+#nullable enable
+
+using System;
+using nadena.dev.modular_avatar.core.vertex_filters;
+using nadena.dev.ndmf.preview;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace nadena.dev.modular_avatar.core.editor
+{
+    [ProvidesVertexFilter(typeof(VertexFilterByBoneComponent))]
+    internal sealed class VertexFilterByBone : IVertexFilter
+    {
+        private readonly Transform _bone;
+        private readonly float _threshold;
+        private readonly Transform _avatarRoot;
+
+        public VertexFilterByBone(Transform bone, float threshold)
+        {
+            _bone = bone;
+            _threshold = threshold;
+            _avatarRoot = null;
+        }
+        
+        public VertexFilterByBone(VertexFilterByBoneComponent component, ComputeContext context)
+        {
+            (_bone, _threshold) = context.Observe(component, c => (c.Bone, c.Threshold));
+            _avatarRoot = RuntimeUtil.FindAvatarTransformInParents(component.transform);
+        }
+
+        public bool Equals(IVertexFilter other)
+        {
+            return other is VertexFilterByBone filter
+                   && filter._bone == _bone
+                   && Mathf.Approximately(filter._threshold, _threshold);
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(typeof(VertexFilterByBone), _bone, _threshold);
+        }
+
+        public override string ToString()
+        {
+            return $"VertexFilterByBone: {(_bone != null ? _bone.name : "null")} @ {_threshold}";
+        }
+
+        public void MarkFilteredVertices(Renderer renderer, Mesh mesh, bool[] filtered)
+        {
+            if (_bone == null || !(renderer is SkinnedMeshRenderer smr))
+            {
+                return; // No bone specified or not a skinned mesh
+            }
+
+            // Find the bone index in the renderer's bones array
+            int boneIndex = -1;
+            for (int i = 0; i < smr.bones.Length; i++)
+            {
+                if (smr.bones[i] == _bone)
+                {
+                    boneIndex = i;
+                    break;
+                }
+            }
```

Comment:
The same bone can appear in the bones array multiple times; use all of them.
Additionally, if a bone is replaced with a generated bone, we must track this as well. As such, you should match `bone` if `ObjectRegistry.GetReference(bone) == ObjectRegistry.GetReference(_bone)`.